src/ca/mcgill/ecse211/lab4/UltrasonicLocalizer.java

```diff
@@ -0,0 +1,262 @@
+
+
+/**
+ * This class implements the first step in the localization process, that is to say ultrasonic
+ * localization. Essentially, it implements the ultrasonic sensor to filter distances through either
+ * the falling edge or the raising edge process. Then, once the filtering is done, two angles (alpha
+ * and beta), and the robot uses those to orient its heading in the direction of increasing y.
+ * 
+ * @author1 Cristian Ciungu
+ * @author2 Hao Shu
+ * @version 12-02-2019
+ * 
+ * 
+ */
+
+// import package
+package ca.mcgill.ecse211.lab4;
+
+import lejos.hardware.Sound;
+import lejos.hardware.motor.EV3LargeRegulatedMotor;
+import lejos.robotics.SampleProvider;
+import ca.mcgill.ecse211.odometer.*;
+import java.util.Arrays;
+import ca.mcgill.ecse211.lab4.*;
+
+public class UltrasonicLocalizer {

```

Comment:
The reason I made everything `static` (see other pull request) is because you only have ONE of each of these: `UltrasonicLocalizer`, `LightLocalizer`, `Navigation`, ie you don't have many instances of these objects. Making everything `static` makes it easier to organize your code. In the context of this project, using static imports makes your code _much_ easier to read, however, avoid them in industry.